front/src/pages/Home/Home.tsx

```diff
@@ -26,23 +27,32 @@ const Home = (): JSX.Element => {
           }));
         }
       },
+      refetchOnWindowFocus: false,
     }
   );
 
-  if (isFetching) {
-    return <Loader />
+  const handleTodoClick = (id: string, isDone: boolean) => {
+    todosAPI.updateTodo({ id, isDone })
```

Comment:
while you approach should work(update + refetch), it's more convenient and actually idiomatic to use `useMutation` if working with react-query

https://tanstack.com/query/v4/docs/guides/optimistic-updates gives example and some details

Reply:
@skyboyer 
Cool feature. But it takes much more code :-)